tooling/miniutil/src/build/mod.rs

```diff
@@ -261,6 +261,44 @@ impl FunctionBuilder {
         self.args.push(name);
         local_by_name(name)
     }
+
+    pub fn cleanup<F>(&mut self, cleanup_builder: F) -> BbName
+    where
+        F: Fn(&mut Self),
+    {
+        let mut cur_block = self.cur_block.take();
+        let cleanup_block = self.declare_block();
+        self.set_cur_block(cleanup_block, BbKind::Cleanup);
+        cleanup_builder(self);
+
+        if self.cur_block.is_some() {
+            panic!("The cleanup block is unfinished. The block needs to end with a Terminator.");
+        }
+        self.cur_block = cur_block.take();
+        cleanup_block
+    }
+
+    pub fn cleanup_resume(&mut self) -> BbName {
+        self.cleanup(|f| {
+            f.resume_unwind();
+        })
+    }
+
+    pub fn terminate<F>(&mut self, terminat_builer: F) -> BbName
```

Comment:
`b.terminate(...)` looks confusing. Maybe call it `terminating_block`?

And then call the other one `cleanup_block` for consistency.

Also please add a doc comment here.